From: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
Subject: man page for pip_stress

The Debian package checker lintian criticizes that pip_stress doesn't have a
man page. So put a short version of the description from the source file
into a man page to make lintian happy.

Signed-off-by: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
---
 Makefile                  |    1 +
 src/pi_tests/pip_stress.8 |   25 +++++++++++++++++++++++++
 2 files changed, 26 insertions(+)

--- /dev/null
+++ b/src/pi_tests/pip_stress.8
@@ -0,0 +1,25 @@
+.TH pip_stress 8 2011-06-24
+.SH NAME
+pip_stress \- test priority inheritance between processes
+.SH SYNOPSIS
+.B pip_stress
+.SH DESCRIPTION
+.B pip_stress
+creates a priority inversion using three processes and a priority inheritance
+mutex shared via shared memory.
+All three processes are bound to run on the same cpu.
+The process with the lowest priority holds a mutex and is prempted by the
+medium priority process that just runs an infinite loop.
+The third process with the highest priority tries to grab the mutex that is
+hold by the low prio process.
+As a priority inheritance mutex is used the low priority process lends the high
+priority process' priority to unlock the mutex and so stops the medium priority
+process blocking the high priority one.
+
+.B pip_stress
+doesn't take any options and quits as soon as the priority inversion is
+resolved which should happen instantly.
+
+.SH AUTHOR
+.B pip_stress
+was written by John Kacur <jkacur@redhat.com>.
--- a/Makefile
+++ b/Makefile
@@ -128,6 +128,7 @@
 	gzip src/backfire/backfire.4 -c >"$(DESTDIR)$(mandir)/man4/backfire.4.gz"
 	gzip src/cyclictest/cyclictest.8 -c >"$(DESTDIR)$(mandir)/man8/cyclictest.8.gz"
 	gzip src/pi_tests/pi_stress.8 -c >"$(DESTDIR)$(mandir)/man8/pi_stress.8.gz"
+	gzip src/pi_tests/pip_stress.8 -c >"$(DESTDIR)$(mandir)/man8/pip_stress.8.gz"
 	gzip src/hwlatdetect/hwlatdetect.8 -c >"$(DESTDIR)$(mandir)/man8/hwlatdetect.8.gz"
 	gzip src/ptsematest/ptsematest.8 -c >"$(DESTDIR)$(mandir)/man8/ptsematest.8.gz"
 	gzip src/sigwaittest/sigwaittest.8 -c >"$(DESTDIR)$(mandir)/man8/sigwaittest.8.gz"
